### ISSUE TITLE
Doesn't read configuration values from environment

### ISSUE THREAD

**Issue by hhovhann** [quarkusio/quarkus#16531]
## Environment
Quarkus version 1.13.1.Final
Java version 11

## Describe the bug
Hi there, thanks for your hard job.

Wondering that something was changed or broken in the new version regarding some default configurations and reading from the environment.
 
### Expected behavior
The email was successfully sent

### Actual behavior
Email notification sent was failed: java.lang.RuntimeException: Error injecting io.vertx.mutiny.ext.mail.MailClient io.quarkus.mailer.runtime.MutinyMailerImpl.client

## To Reproduce

https://github.com/hhovhann/code-with-quarkus-mailer

1.  Run /scripts/init_email_test_env_variables.sh for init application configurations before running app
2.  http://localhost:8080/q/swagger-ui/ -> Send POST /notification/send-email
3. Run `mvn clean compile quarkus:dev`

**Comment by hhovhann** [comment]
FYI: If you are using the script, it should be run via `source /scripts/init_email_test_env_variables.sh`, which works and init variables and email will be sent.
Tried also with the `.env` and added there the values but after running application via `mvn clean compile quarkus:dev` it gives an exception
```
Listening for transport dt_socket at address: 5005
2021-04-15 15:52:39,390 INFO  [io.qua.ely.sec.pro.dep.ElytronPropertiesProcessor] (build-52) Configuring from MPRealmConfig
2021-04-15 15:52:40,538 INFO  [io.qua.ver.dep.EventBusCodecProcessor] (build-54) Local Message Codec registered for type java.util.Map
2021-04-15 15:52:40,540 INFO  [io.qua.ver.dep.EventBusCodecProcessor] (build-54) Local Message Codec registered for type org.modelmapper.internal.Pair

   _____       __                       ____  ____  ____
  / ___/____  / /_  ___  _________ _   / __ \/ __ \/ __ \
  \__ \/ __ \/ __ \/ _ \/ ___/ __ `/  / / / / /_/ / /_/ /
 ___/ / /_/ / / / /  __/ /  / /_/ /  / /_/ / _, _/ ____/
/____/ .___/_/ /_/\___/_/   \__,_/  /_____/_/ |_/_/
    /_/

                          Powered by Quarkus 1.13.1.Final
2021-04-15 15:53:17,165 WARN  [io.agr.pool] (agroal-11) Datasource 'common': The TCP/IP connection to the host localhost, port 1434 has failed. Error: "connect timed out. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.".
2021-04-15 15:53:17,235 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile drp): com.microsoft.sqlserver.jdbc.SQLServerException: The TCP/IP connection to the host localhost, port 1434 has failed. Error: "connect timed out. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.".
        at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:234)
        at com.microsoft.sqlserver.jdbc.SQLServerException.ConvertConnectExceptionToSQLServerException(SQLServerException.java:285)
        at com.microsoft.sqlserver.jdbc.SocketFinder.findSocket(IOBuffer.java:2431)
        at com.microsoft.sqlserver.jdbc.TDSChannel.open(IOBuffer.java:656)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectHelper(SQLServerConnection.java:2440)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.login(SQLServerConnection.java:2103)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:1950)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:1162)
        at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:735)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:200)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:452)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:434)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:65)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)

2021-04-15 15:53:17,238 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
Quarkus application exited with code 1
Press Enter to restart or Ctrl + C to quit
2021-04-15 15:53:17,307 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.nio.file.NoSuchFileException: server.keystore
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:228)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:37)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:653)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:137)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:378)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:127)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: java.nio.file.NoSuchFileException: server.keystore
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:421)
        at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)
        at java.base/java.nio.file.Files.newInputStream(Files.java:155)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.getFileContent(VertxHttpRecorder.java:670)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.createSslOptions(VertxHttpRecorder.java:612)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:468)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:220)
        ... 9 more
```

**Comment by radcortez** [comment]
Hi @hhovhann

I've tried the reproducer and added my own `.env` file with the same configuration names, but pointing to a personal email server and I has able to send emails.

The included stacktrace seems to point to another issue, related with a database connection and/or missing keystores. The reproducer project does not include any database connection, so I supposed the stacktrace is not from the reproducer.

This is what I get with the reproducer:

```console
2021-04-19 15:59:31,579 INFO  [io.quarkus] (Quarkus Main Thread) code-with-quarkus-mailer 1.0.0-SNAPSHOT on JVM (powered by Quarkus 1.13.1.Final) started in 1.579s. Listening on: http://localhost:8080
2021-04-19 15:59:31,584 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-04-19 15:59:31,584 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, mailer, mutiny, qute, resteasy, smallrye-context-propagation, smallrye-openapi, swagger-ui, vertx]
2021-04-19 15:59:39,930 INFO  [com.hho.ser.NotificationService] (vert.x-eventloop-thread-2) Email notification was successfully sent.
^C2021-04-19 15:59:45,259 INFO  [io.quarkus] (Shutdown thread) code-with-quarkus-mailer stopped in 0.021sls
```

**Comment by hhovhann** [comment]
Thanks for the clarification and checks.
